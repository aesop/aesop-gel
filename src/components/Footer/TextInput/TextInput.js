import React from 'react';
import PropTypes from 'prop-types';
import styles from './TextInput.module.css';
import cx from 'classnames';

const TextInput = ({
  className,
  defaultValue,
  disabled,
  errorMessage,
  hasContent,
  id,
  inputClassName,
  inputRef,
  label,
  max,
  min,
  name,
  type,
  maxLength,
  novalidate,
  onBlur,
  onChange,
  prefixElement,
  value,
  ...moreProps
}) => {
  const dataTestRef = moreProps['data-test-ref'];
  const classNames = cx(styles.formText, className);
  const inputClassNames = cx(
    styles.formTextInput,
    (!!defaultValue || hasContent) && styles.hasContent,
    !!errorMessage && styles.hasError,
    inputClassName,
  );

  return (
    <div className={classNames}>
      <label htmlFor={id}>
        <input
          className={inputClassNames}
          data-test-ref={dataTestRef}
          defaultValue={defaultValue}
          id={id}
          disabled={disabled}
          max={max}
          maxLength={maxLength}
          min={min}
          name={name}
          noValidate={novalidate}
          onBlur={onBlur}
          onChange={onChange}
          placeholder=""
          ref={inputRef}
          type={type || 'text'}
          value={value}
        />
        <span className={styles.formTextLabel}>{label}</span>
        {prefixElement}
      </label>
      {errorMessage && (
        <div className={styles.formTextErrorMessage}>{errorMessage}</div>
      )}
    </div>
  );
};

TextInput.propTypes = {
  className: PropTypes.string,
  defaultValue: PropTypes.string,
  disabled: PropTypes.bool,
  errorMessage: PropTypes.string,
  hasContent: PropTypes.bool,
  id: PropTypes.string,
  inputClassName: PropTypes.string,
  inputRef: PropTypes.func,
  label: PropTypes.string,
  max: PropTypes.number,
  min: PropTypes.number,
  name: PropTypes.string,
  maxLength: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  novalidate: PropTypes.bool,
  prefixElement: PropTypes.element,
  type: PropTypes.oneOf([
    'text',
    'password',
    'email',
    'number',
    'tel',
    'search',
  ]),
  'data-test-ref': PropTypes.string,
  value: PropTypes.string,
};

export default TextInput;
