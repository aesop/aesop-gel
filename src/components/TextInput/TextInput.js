import React from 'react';
import PropTypes from 'prop-types';
import styles from './TextInput.module.css';
import cx from 'classnames';

const TextInput = ({
  className,
  defaultValue,
  disabled,
  errorMessage,
  hasContent,
  id,
  inputClassName,
  inputRef,
  label,
  max,
  min,
  name,
  type,
  maxLength,
  novalidate,
  onBlur,
  onChange,
  prefixElement,
  theme,
  value,
  ...moreProps
}) => {
  const dataTestRef = moreProps['data-test-ref'];
  const classNames = cx(styles.formText, className);
  const inputClassNames = cx(
    styles.formTextInput,
    styles[theme],
    (!!defaultValue || hasContent) && styles.hasContent,
    !!errorMessage && styles.hasError,
    inputClassName,
  );
  const labelClassSet = cx(styles.formTextLabel, styles[theme]);

  return (
    <div className={classNames}>
      <label htmlFor={id}>
        <input
          className={inputClassNames}
          data-test-ref={dataTestRef}
          defaultValue={defaultValue}
          disabled={disabled}
          id={id}
          max={max}
          maxLength={maxLength}
          min={min}
          name={name}
          noValidate={novalidate}
          onBlur={onBlur}
          onChange={onChange}
          placeholder=""
          ref={inputRef}
          type={type || 'text'}
          value={value}
        />
        <span className={labelClassSet}>{label}</span>
        {prefixElement}
      </label>
      {errorMessage && (
        <div className={styles.formTextErrorMessage}>{errorMessage}</div>
      )}
    </div>
  );
};

TextInput.propTypes = {
  className: PropTypes.string,
  defaultValue: PropTypes.string,
  disabled: PropTypes.bool,
  errorMessage: PropTypes.string,
  hasContent: PropTypes.bool,
  id: PropTypes.string.isRequired,
  inputClassName: PropTypes.string,
  inputRef: PropTypes.func,
  label: PropTypes.string,
  max: PropTypes.number,
  min: PropTypes.number,
  name: PropTypes.string,
  maxLength: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  novalidate: PropTypes.bool,
  prefixElement: PropTypes.element,
  theme: PropTypes.oneOf(['dark', 'light']),
  type: PropTypes.oneOf([
    'text',
    'password',
    'email',
    'number',
    'tel',
    'search',
  ]),
  'data-test-ref': PropTypes.string,
  value: PropTypes.string,
};

export default TextInput;
