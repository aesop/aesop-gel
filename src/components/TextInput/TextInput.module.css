@import 'styles/index';

.formText {
  width: 100%;
}

.formTextInput {
  position: relative;
  width: 100%;
  box-sizing: border-box;
  padding-left: 10px;
  border: 1px solid var(--color-white);
  background-color: transparent;
  border-radius: 3px;
  caret-color: var(--color-white);
  color: var(--color-white);
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(20px);
  line-height: 1.85;
  &.hasError {
    border-color: red;
  }

  &:focus {
    outline: none;
  }

  /* Overrides for autofill coloring */
  &:-webkit-autofill,
  &:-webkit-autofill:hover,
  &:-webkit-autofill:focus {
    -webkit-text-fill-color: var(--color-white);
    transition: background-color 5000s ease-in-out 0s;
  }
}

.formTextLabel {
  position: absolute;
  top: 2px;
  left: 10px;
  color: var(--color-white);
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(20px);
  line-height: 1.85;
  opacity: 1;
  pointer-events: none;
  transition: all 300ms var(--easing-ease-in-out-cubic);

  &.is-hidden {
    opacity: 0;
  }

  .formTextInput:focus + &,
  .formTextInput.hasContent + & {
    opacity: 0;
  }
}

.formTextErrorMessage {
  color: red;
}
