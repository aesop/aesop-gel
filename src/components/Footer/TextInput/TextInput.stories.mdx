import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import Podium from '~/components/Podium';
import TextInput from './TextInput';

<Meta title="Components/TextInput" component={TextInput} />

# TextInput

<Preview>
  <Story name="Base component">
    <Podium horizontalPadding="small" verticalPadding="small">
      <p>Set the background to dark</p>
      <TextInput name='test-input' />
    </Podium>
  </Story>
</Preview>

## API

<Props of={TextInput} />
