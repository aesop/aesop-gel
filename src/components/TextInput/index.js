/* istanbul ignore file */
import { TextInput } from './TextInput';

export { TextInput };
