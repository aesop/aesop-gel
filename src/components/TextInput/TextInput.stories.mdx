import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { Podium } from '~/components/Podium';
import { TextInput } from './TextInput';

<Meta title="Form Elements/TextInput" component={TextInput} />

# TextInput

<Canvas>
  <Story name="Base component">
    <Podium horizontalPadding="small" verticalPadding="small">
      <TextInput
        id="test-input"
        theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
      />
    </Podium>
  </Story>
</Canvas>

## API

<ArgsTable of={TextInput} />
