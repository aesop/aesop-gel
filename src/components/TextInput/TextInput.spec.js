import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';

configure({ adapter: new Adapter() });

describe('<TextInput />', () => {
  it('should be defined', () => {
    expect(TextInput).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(<TextInput data-test-ref="test-data-ref" id="test-textinput" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
