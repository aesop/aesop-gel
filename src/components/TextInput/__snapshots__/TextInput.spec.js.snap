// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders base component correctly 1`] = `
<div
  className="formText"
>
  <label
    htmlFor="test-textinput"
  >
    <input
      className="formTextInput undefined"
      id="test-textinput"
      placeholder=""
      type="text"
    />
    <span
      className="formTextLabel undefined"
    />
  </label>
</div>
`;
